numpy/lib/tests/test_type_check.py

```diff
@@ -320,6 +320,15 @@ def test_generic(self):
         assert_(vals[1] == 0)
         assert_all(vals[2] > 1e10) and assert_all(np.isfinite(vals[2]))
 
+        # perform the same test but in-place
+        with np.errstate(divide='ignore', invalid='ignore'):
+            vals = np.array((-1., 0, 1))/0.
+        nan_to_num(vals, copy=False)
```

Comment:
Would be good to test the return value here too, and that it is just the same as the argument